lib/ui/geometry.dart

```diff
@@ -1018,81 +1004,78 @@ class Radius {
 
 /// An immutable rounded rectangle with the custom radii for all four corners.
 class RRect {
-  RRect._();
-
   /// Construct a rounded rectangle from its left, top, right, and bottom edges,
   /// and the same radii along its horizontal axis and its vertical axis.
-  RRect.fromLTRBXY(double left, double top, double right, double bottom,
-                   double radiusX, double radiusY) {
-    _value
-      ..[0] = left
-      ..[1] = top
-      ..[2] = right
-      ..[3] = bottom
-      ..[4] = radiusX
-      ..[5] = radiusY
-      ..[6] = radiusX
-      ..[7] = radiusY
-      ..[8] = radiusX
-      ..[9] = radiusY
-      ..[10] = radiusX
-      ..[11] = radiusY;
-  }
+  const RRect.fromLTRBXY(double left, double top, double right, double bottom,
+                   double radiusX, double radiusY) : this._raw(
+    top: top,
+    left: left,
+    right: right,
+    bottom: bottom,
+    tlRadiusX: radiusX,
+    tlRadiusY: radiusY,
+    trRadiusX: radiusX,
+    trRadiusY: radiusY,
+    blRadiusX: radiusX,
+    blRadiusY: radiusY,
+    brRadiusX: radiusX,
+    brRadiusY: radiusY,
+  );
```

Comment:
rather than a ._raw constructor you could also just inline the initializers in each constructor

Reply:
The `_raw` constructor asserts everything is not null.